lib/Dialect/MSFT/MSFTPasses.cpp

```diff
@@ -502,6 +502,122 @@ static StringRef getOperandName(OpOperand &oper, const hw::SymbolCache &syms,
   return buff;
 }
 
+/// Helper to update GlobalRefOps after referenced ops bubble up.
+static void bubbleUpGlobalRefs(Operation *op) {
+  if (auto globalRefs = op->getAttrOfType<ArrayAttr>("circt.globalRef")) {
+    for (auto globalRef : globalRefs.getAsRange<hw::GlobalRefAttr>()) {
+      // Resolve the GlobalRefOp and get its path.
+      auto rootModule = op->getParentOfType<ModuleOp>();
+      auto refSymbol = globalRef.getGlblSym();
+      auto globalRefOp = rootModule.lookupSymbol<hw::GlobalRefOp>(refSymbol);
+      auto oldPath = globalRefOp.namepath().getAsRange<hw::InnerRefAttr>();
+
+      // Construct a new path starting from the old path.
+      SmallVector<Attribute> newPath(oldPath);
+
+      // If there aren't two elements in the path, this GlobalRefOp has already
+      // been updated.
+      if (newPath.size() < 2)
+        return;
+
+      // Splice the last two elements in the path, combining their names with a
+      // dot, and taking the module from the second to last element.
+      auto lastElement = newPath.pop_back_val().cast<hw::InnerRefAttr>();
+      auto nextElement = newPath.pop_back_val().cast<hw::InnerRefAttr>();
+      auto newModule = nextElement.getModule();
+
+      SmallString<16> newName;
+      newName += nextElement.getName().getValue();
```

Comment:
Does this have to correspond with the moved op's new symbol? If so, you shouldn't duplicate that functionality. Maybe accept the new name as a function parameter. Or read the 'inner_sym' attribute from op.